src/python/pants/backend/python/lint/pylint/rules_integration_test.py

```diff
@@ -0,0 +1,187 @@
+      args.append("--pylint-config=pylintrc")
+    if passthrough_args:
+      args.append(f"--pylint-args='{passthrough_args}'")
+    if skip:
+      args.append(f"--pylint-skip")
+    input_snapshot = self.request_single_product(Snapshot, InputFilesContent(source_files))
+    target = PylintTarget(
+      PythonTargetAdaptor(
+        sources=EagerFilesetWithSpec(self.source_root, {'globs': []}, snapshot=input_snapshot),
+        address=Address.parse(f"{self.source_root}:target"),
+        compatibility=[interpreter_constraints] if interpreter_constraints else None,
+      )
+    )
+    return self.request_single_product(
+      LintResult, Params(target, create_options_bootstrapper(args=args)),
+    )
+
+  def test_single_passing_source(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.good_source)
+    result = self.run_pylint([self.good_source])
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_single_failing_source(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source])
+    assert result.exit_code == 16
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  def test_mixed_sources(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.good_source)
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.good_source, self.bad_source])
+    assert result.exit_code == 16
+    assert "good.py" not in result.stdout
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  @pytest.mark.skip(reason="Get config file creation to work with options parsing")
+  def test_respects_config_file(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source], config="[pylint]\ndisable = C0114\n")
+    assert result.exit_code == 0
+    assert result.stdout.strip() == ""
+
+  def test_respects_passthrough_args(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source], passthrough_args="--disable=C0114")
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_transitive_dep(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="library")
+    self.create_library(
+      path=self.source_root, target_type="python_library", name="transitive_dep",
+      sources=["transitive_dep.py"], dependencies=[":library"],
+    )
+    self.write_file(
+      FileContent(
+        path="transitive_dep.py",
```

Comment:
Can rename to `dependency.py`. The transitive part is a misnomer as this is directly used by the following file.